akka-http-tests/src/test/scala/akka/http/scaladsl/marshalling/ContentNegotiationSpec.scala

```diff
@@ -136,6 +136,40 @@ class ContentNegotiationSpec extends FreeSpec with Matchers {
       accept(`text/plain`, `text/xml`) should select(`text/plain` withCharset `UTF-8`)
       accept(`text/xml`, `text/plain`) should select(`text/xml` withCharset `UTF-8`)
     }
+
+    /** when both acceptable media types are equally specific and have the same qValue, we choose to send the first */
+    "Accept: text/html;level=1, text/html;format=flowed" test { accept ⇒
+      accept(`text/html` withParams Map("level" → "1"), `text/html` withParams Map("format" → "flowed")) should select(`text/html` withParams Map("level" → "1") withCharset `UTF-8`)
+    }
+
+    /** when an acceptable media type is more specific, it is matched even when it has a lower qValue */
```

Comment:
`text/html;format=flowed` is more specific than `text/html` because it specifies (restricts) more parameters.

In this example, we have 2 marshallers available, one with media type `text/html;level=1` and one with media type `text/html;format=flowed`. For each one we calculate the qValue based on the preferences in the `Accept` header. 

The `text/html;level=1` marshalling does not match `text/html;format=flowed` because it is missing the `format` parameter, but does match `text/html`, so it gets the associated qValue of 1. 

The `text/html;format=flowed` marshalling matches both `text/html;format=flowed` and `text/html`. Because `text/html;format=flowed` is more specific, this one is selected, and a qValue of 0.5 is assigned to this marshalling.

Now that we have assigned the qValues, we select the marshalling with the largest qValue, which is `text/html;level=1`.

This is my interpretation of https://tools.ietf.org/html/rfc7231#section-5.3.2 , though I agree it's not too easy to read.